src/meta/client/MetaClient.cpp

```diff
@@ -286,6 +286,31 @@ MetaClient::reverse(const PartsAlloc& parts) {
     return hosts;
 }
 
+template<typename Request,
+         typename RemoteFunc,
+         typename RespGenerator,
+         typename RpcResponse,
+         typename Response>
+void MetaClient::retry(Request req,
+                       RemoteFunc remoteFunc,
+                       RespGenerator respGen,
+                       folly::Promise<StatusOr<Response>> pro,
+                       bool toLeader,
+                       int32_t retry,
+                       int32_t retryLimit) {
+    folly::EventBase retryEvb;
```

Comment:
Why not reuse current evb inside ioThreadPool?

Reply:
I've tried before. Quoted from folly:
>   Your code just tried to loop over an event base from inside another event base loop. Since libevent is not reentrant, this leads to undefined behavior in opt builds. Please fix immediately. For the common case of an inner function that needs to do some synchronous computation on an event-base, replace getEventBase() by a new, stack-allocated EvenBase.

Reply:
> I've tried before. Quoted from folly:
> 
> > Your code just tried to loop over an event base from inside another event base loop. Since libevent is not reentrant, this leads to undefined behavior in opt builds. Please fix immediately. For the common case of an inner function that needs to do some synchronous computation on an event-base, replace getEventBase() by a new, stack-allocated EvenBase.

Actually, by reusing the present evb, you dont loop by yourself, you are just to add a timer event to the current event loop.

Reply:
> Actually, by reusing the present evb, you dont loop by yourself, you are just to add a timer event to the current event loop.

Yeah, but when I reuse the present evb (aka evb from ioThreadPool), the retry timer event was never fired, it really confuses me.

Reply:
You could write the code in this way 
```
evb->runAfterDelay([req = std::move(req),
                                    remoteFunc = std::move(remoteFunc),
                                    respGen = std::move(respGen),
                                    pro = std::move(pro),
                                    toLeader,
                                    retry,
                                    retryLimit,
                                    this]) mutable {
      getResponse(std::move(req), 
                             std::move(remoteFunc),
                             std::move(respGen),
                             std::move(pro),
                             toLeader,
                             retry + 1,
                             retryLimit);
},  FLAGS_meta_client_retry_interval_secs * 1000);
```